capellambse/svg/drawing.py

```diff
@@ -970,11 +971,11 @@ def get_label_position(
     assert builder.labels
     assert lines
 
+    label_height = chelpers.extent_func(builder.labels[0]["text"])[1]
+    dominant_baseline_adjust = label_height / 2
     for label in builder.labels:
-        label_height = chelpers.extent_func(label["text"])[1]
-        dominant_baseline_adjust = label_height / 2
         if builder.text_anchor == "start":
-            label["x"] += lines.margin + icon_size
+            label["x"] += lines.margin + icon_size / 2
```

Comment:
This possibly has an effect on other diagrams. As labels are one of the biggest pain points for the diagram rendering engine, extra caution is needed here. The changes done here do:
* fixate the label_height from the first line of the label (from dynamic to static line_height)
* Reduce the distance of the icon to the label itself, by half of the icon_size (which is normally 25x25 therefore 12.5)

Did you check if other diagram type rendering is still ok?

Reply:
1. Since we calculate `label_height` with `chelpers.extent_func` instead of `chelpers.get_text_extent` and all labels of a `LabelBuilder` object have the same font size, shouldn't `label_height` always equal to the same value? Maybe we should use `chelpers.get_text_extent` instead then to account for potential line wrapping.
2. This doesn't reduce the distance of the icon to the label, as icon position is calculated relative to label position, but the distance of the box's left border to the label.
I don't know the reason but even when leaving `label["x"]` as is there is a giant gap between the label and the box's border.
I haven't checked with other diagram types as I don't really know any other diagram types where `text_anchor` defaults to "start".

Reply:
> 1. Since we calculate `label_height` with `chelpers.extent_func` instead of `chelpers.get_text_extent` and all labels of a `LabelBuilder` object have the same font size, shouldn't `label_height` always equal to the same value? Maybe we should use `chelpers.get_text_extent` instead then to account for potential line wrapping.

No. The `label_height` is determined by 3 different values: Baseline, upper and lower extensions (there are dedicated terms for this, which I can't remember). So for example:

`acore` is a word which has only the baseline in its height, `bookstore` has baseline and the upper extension through `b`, `k` and `t`. `Any` has a lower extension through `y`, an upper by `A` and baseline as always. So the `line_height` should be determined by iterating through all letters and picking up the maximum height of each glyph... We are using the font measure functionality of Pillow with the `getsize` method on FreeFontType. [Looking into the docs](https://pillow.readthedocs.io/en/stable/releasenotes/10.0.0.html#font-size-and-offset-methods),... this way isn't supported anymore and we should use a different method: `getbbox` or `getlength`. 

>     2. This doesn't reduce the distance of the icon to the label, as icon position is calculated relative to label position, but the distance of the box's left border to the label.
>        I don't know the reason but even when leaving `label["x"]` as is there is a giant gap between the label and the box's border.
>        I haven't checked with other diagram types as I don't really know any other diagram types where `text_anchor` defaults to "start".

I have no clue too. I wrote the logic, but I can't remember without reading the lines of code.

Reply:
As Ernst described, there can be very subtle differences in height between the different labels due to the letters used. So it makes sense to look at each label individually, instead of taking the first label for all the others as well.

I've also tested across a number of diagrams from one of our larger internal models, and there have indeed been differences in height (although expectedly very small, during the spot-check I didn't see anything larger than 3 pixels in total). Still, the "right thing" here IMO is to look at all the labels as individual components.

Other than this, the PR looks good to me, so I'm just going to change this bit myself and merge.